errai-security/errai-security-picketlink/src/main/java/org/jboss/errai/security/server/PicketLinkAuthenticationService.java

```diff
@@ -61,11 +58,14 @@
   private Identity identity;
 
   @Inject
-  private RelationshipManager relationshipManager;
+  private PicketLinkBasicModelServices basicModelServices;
 
   @Inject
   private DefaultLoginCredentials credentials;
 
+  @Inject
+  private RequiredRolesProvider requiredRolesProvider;
+
```

Comment:
Unfortunately, injecting a RequiredRolesProvider here won't work. It was mainly designed for the client so roles can be provided at runtime, see RestrictedAccess(provider=...). In fact, this will actually cause deployment errors for all applications that don't provide an implementation of that interface! We definitely do not want that.